packages/console/src/pages/Callback/index.tsx

```diff
@@ -1,12 +1,24 @@
 import { LogtoError, OidcError, useHandleSignInCallback } from '@logto/react';
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 import AppError from '@/components/AppError';
 import LogtoLoading from '@/components/LogtoLoading';
 import { getBasename } from '@/utilities/app';
 
 const Callback = () => {
-  const { error } = useHandleSignInCallback(getBasename());
+  const { error, isAuthenticated } = useHandleSignInCallback(getBasename());
+  const navigate = useNavigate();
+
+  /**
+   * Redirect back to the home page if the user is already authenticated.
+   * Corner case when user mistakenly navigate to `/callback` route after a successful authentication.
+   */
+  useEffect(() => {
+    if (isAuthenticated) {
```

Comment:
Should we also check error here? I'm fine with both

Reply:
Not needed. The error will always be `undefined` in this case.

Because if the user is authenticated, it will not trigger the `handleSignInCallback()` logic whatsoever. Hence there will be nothing happening on this page, nor error.